### PR TITLE
Adds a helper for django url tests

### PR DESCRIPTION
**Pull request by asmacdo**
This base class allows significantly simpler tests for each django url.

### CODE REVIEW

File: server/test/unit/server/webservices/test_urls.py [hunk 1]
```diff
@@ -3,6 +3,40 @@
...
+class TestDjangoUrls(unittest.TestCase):
+    """
+    Base class for testing Django Urls.
+    """
+    def assert_url_match(self, urls, url_name, match=True, *args):
+        """
+        Asserts that a url matches the url_name if match == True,
+        asserts that a url does not match the url_name if match==False.
+
+        :param url     : url or urls to match or base url to build from
```
Comment (bmbouter) [inline server/test/unit/server/webservices/test_urls.py]: The param in the signature is 'urls' (with the s) but this says 'url' is the param name.

File: server/test/unit/server/webservices/test_urls.py [hunk 2]
```diff
@@ -3,6 +3,40 @@
...
+        :type  url     : string or list of strings
+        :param url_name: expected name of the url match
+        :type  url_name: string
+        :param match   : If true, check that urls match, if false, check that urls do not match
```
Comment (bmbouter) [inline server/test/unit/server/webservices/test_urls.py]: missing a type for 'match' param

File: server/test/unit/server/webservices/test_urls.py [hunk 3]
```diff
@@ -3,6 +3,40 @@
...
+        :args          : optional arguments to append to the end of the url
```
Comment (bmbouter) [inline server/test/unit/server/webservices/test_urls.py]: :args is not a recognized sphinx command. I think :param args maybe is what you want.

:args is also missing a :type describing its type

File: server/test/unit/server/webservices/test_urls.py [hunk 4]
```diff
@@ -1,6 +1,35 @@
 import unittest
 
-from django.core.urlresolvers import resolve, Resolver404
+from django.core.urlresolvers import resolve, reverse, NoReverseMatch
+
+
+def assert_url_match(expected_url, url_name, *args, **kwargs):
+        """
+        Generate a url given args and kwargs and pass it though Django's reverse and
```
Comment (mhrivnak) [inline server/test/unit/server/webservices/test_urls.py]: s/though/through

File: server/test/unit/server/webservices/test_urls.py [hunk 5]
```diff
@@ -1,6 +1,35 @@
...
+            url = reverse(url_name, args=args, kwargs=kwargs)
+        except NoReverseMatch:
+            raise AssertionError(
+                "Name: '{0}' could match a url with args '{1}'"
+                "and kwargs '{2}'".format(url_name, args, kwargs)
+            )
+
+        else:
+            # If the url exists but is not the expected url.
+            if not url == expected_url:
```
Comment (mhrivnak) [inline server/test/unit/server/webservices/test_urls.py]: This isn't wrong, but it seems simpler and cleaner to just use ``!=``

File: server/test/unit/server/webservices/test_urls.py [hunk 6]
```diff
@@ -1,6 +1,35 @@
...
+                raise AssertionError(
+                    'url {0} not equal to expected url {1}'.format(url, expected_url))
+
+            # Run this url back through resolve and ensure that it matches the url_name.
+            matched_view = resolve(url)
+            if not matched_view.url_name == url_name:
```
Comment (mhrivnak) [inline server/test/unit/server/webservices/test_urls.py]: same here.

File: server/test/unit/server/webservices/test_urls.py [hunk 7]
```diff
@@ -49,3 +78,19 @@ def test_match_repo_group_distributor_resource(self):
...
         pass
+
+
+class TestDjangoRepositoriesUrls(unittest.TestCase):
+
+    def test_match_repositories_resource_sync(self):
+        url = '/v2/repositories/a1b2%21/actions/sync/'
+        url_name = 'repositories_resource_sync'
+        assert_url_match(url, url_name, repo_id='a1b2!')
+
```
Comment (mhrivnak) [inline server/test/unit/server/webservices/test_urls.py]: two blank lines for PEP8

**Comment by asmacdo** [comment]
@mhrivnak ready for rereview

File: server/test/unit/server/webservices/test_urls.py [hunk 8]
```diff
@@ -1,6 +1,46 @@
...
+        else:
+            # If the url exists but is not the expected url.
+            if url != expected_url:
+                raise AssertionError(
+                    'url {0} not equal to expected url {1}'.format(url, expected_url))
+
+            # Run this url back through resolve and ensure that it matches the url_name.
+            matched_view = resolve(url)
+            if matched_view.url_name != url_name:
+                raise AssertionError('url {0} not equal to expected url {1}' % url, expected_url)
```
Comment (mhrivnak) [inline server/test/unit/server/webservices/test_urls.py]: I think everything following the string is incorrect, starting with the ``%``. I think you want to use ``.format``

**Comment by mhrivnak** [comment]
Other than the one last comment, LGTM

**Pull request merged**